docs(datalist): Document labeled values for range inputs

Shrinivassab opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. 🙌

Add details below to help us review your pull request (PR).
Explain your changes and link to a related issue or pull request.
Your PR may be delayed or closed if you don't provide enough information. -->

### Description

<!-- ✍️ Summarize your changes in one or two sentences. -->
Adds documentation showing how to use `label` attributes with `<option>` elements in range-type datalists. Includes:
- Labeled value example in the "Range type" section
- Note about browser tooltip behavior
- Link to relevant HTML spec section

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->

### Additional details

<!-- 🔗 Link to release notes, browser docs, bug trackers, source control, or other resources. -->
**Browser behavior verified in:**
- Chrome 125: Shows labels as hover tooltips
- Firefox 126: Shows labels as hover tooltips
- Safari 17: Shows labels as hover tooltips

**Spec reference:**  
https://html.spec.whatwg.org/multipage/input.html#range-state-(type=range)

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request must be merged first, use "**Depends on:** #123" -->
Fixes #39953

<!-- 🔎 After submitting, the 'Checks' tab of your PR shows the build status. -->

Shrinivassab committed:
docs: Add labeled datalist example for range inputs

chrisdavidmills reviewed (commented):
Hi @Shrinivassab, and thank you for your work on this PR.

Before I review it properly, I've got a comment to start with — from what I've read in the spec, it looks like including label attributes on the `<option>` elements is supposed to make those labels appear below the tickmarks, rather than as tooltips like you say.

I tested both your modified example and the example in the spec on the newest Firefox, Chrome, and Safari, and ... it didn't seem to work in any of them, as a tooltip or a visible label. I tested it on Windows and macOS. Did you definitely see this working?

Shrinivassab committed:
docs(datalist): Clarify label attribute behavior for range inputs

Shrinivassab commented:
Thanks for the review, @chrisdavidmills , great point.

I did initially observe tooltips in a custom testing setup (possibly browser developer tools injected the label values), but you are right, in normal user-facing behavior, labels don't consistently appear as tooltips or visually below tick marks.

I have now updated the **note section** to clarify that while the `label` attribute is defined in the spec for use with range-type inputs and datalists, browser support is inconsistent and the labels may not be displayed visually.

To help developers communicate value meaning to users, we could suggest a manual legend below the slider, like this:
```html
<div style="font-size: 0.9em; margin-top: 0.5em;">
  <strong>Legend:</strong> 0% | Minimum Tip | Standard | Generous | Very Generous
</div>
```

Appreciate the feedback, I will push the update shortly.

Shrinivassab committed:
Merge branch 'main' into range-input-labels

Shrinivassab committed:
Merge branch 'main' into range-input-labels

Shrinivassab commented:
Hi @chrisdavidmills: 
 Just wanted to follow up on this PR when you have a moment. I have incorporated your feedback and updated the note.
I know you are likely busy, but I would really appreciate your review whenever time allows. Thanks again for your thoughtful input earlier.

chrisdavidmills committed:
Section cleanup

chrisdavidmills reviewed (approved):
Thanks, @Shrinivassab. I took this opportunity to do a bit of text cleanup and format the note properly.

I've approved the PR; can you have a look and make sure you are OK with my wording update before we merge it?

chrisdavidmills committed:
Correct GFM note syntax

chrisdavidmills committed:
Merge branch 'main' into range-input-labels
